File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/businessmanagementvb/vb/convertquotetosalesorder.vb
' <snippetconvertquotetosalesorder>


 ' Define columns to be retrieved after creating the order
 Dim salesOrderColumns As New ColumnSet("salesorderid", "totalamount")

 ' Convert the quote to a sales order
 Dim convertQuoteRequest As New ConvertQuoteToSalesOrderRequest() With
     {
         .QuoteId = _quoteId,
         .ColumnSet = salesOrderColumns
     }
 Dim convertQuoteResponse As ConvertQuoteToSalesOrderResponse =
     CType(_serviceProxy.Execute(convertQuoteRequest), 
         ConvertQuoteToSalesOrderResponse)
 Dim salesOrder_Renamed As SalesOrder = CType(convertQuoteResponse.Entity, 
     SalesOrder)
 _salesOrderId = salesOrder_Renamed.Id

 Console.WriteLine("Converted Quote to SalesOrder.")

' </snippetconvertquotetosalesorder>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/businessmanagement/cs/convertquotetosalesorder.cs
// <snippetconvertquotetosalesorder>


 // Define columns to be retrieved after creating the order
 ColumnSet salesOrderColumns =
     new ColumnSet("salesorderid", "totalamount");

 // Convert the quote to a sales order
 ConvertQuoteToSalesOrderRequest convertQuoteRequest =
     new ConvertQuoteToSalesOrderRequest()
     {
         QuoteId = _quoteId,
         ColumnSet = salesOrderColumns
     };
 ConvertQuoteToSalesOrderResponse convertQuoteResponse =
     (ConvertQuoteToSalesOrderResponse)_serviceProxy.Execute(convertQuoteRequest);
 SalesOrder salesOrder = (SalesOrder)convertQuoteResponse.Entity;
 _salesOrderId = salesOrder.Id;

// </snippetconvertquotetosalesorder>